### PR TITLE
refactor(vehicle_external_cmd_converter): rework parameters

### PR DESCRIPTION
**Pull request by karishma1911**
## Description

Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the vehicle_external_cmd_converter

1. Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.

 2.create schema

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by TakaHoribe — APPROVED** [review]
LGTM!

File: vehicle/external_cmd_converter/src/node.cpp [hunk 1]
```diff
@@ -42,13 +42,13 @@ ExternalCmdConverterNode::ExternalCmdConverterNode(const rclcpp::NodeOptions & n
...
   // Parameter
-  ref_vel_gain_ = declare_parameter("ref_vel_gain", 3.0);
+  ref_vel_gain_ = declare_parameter<number>("ref_vel_gain");
 
   // Parameter for Hz check
-  const double timer_rate = declare_parameter("timer_rate", 10.0);
-  wait_for_first_topic_ = declare_parameter("wait_for_first_topic", true);
-  control_command_timeout_ = declare_parameter("control_command_timeout", 1.0);
-  emergency_stop_timeout_ = declare_parameter("emergency_stop_timeout", 3.0);
+  const double timer_rate = declare_parameter<number>("timer_rate");
```
Comment (satoshi-ota) [inline vehicle/external_cmd_converter/src/node.cpp:48]:
```suggestion
  const double timer_rate = declare_parameter<double>("timer_rate");
```

File: vehicle/external_cmd_converter/src/node.cpp [hunk 2]
```diff
@@ -42,13 +42,13 @@ ExternalCmdConverterNode::ExternalCmdConverterNode(const rclcpp::NodeOptions & n
     std::bind(&ExternalCmdConverterNode::onEmergencyStopHeartbeat, this, _1));
 
   // Parameter
-  ref_vel_gain_ = declare_parameter("ref_vel_gain", 3.0);
+  ref_vel_gain_ = declare_parameter<number>("ref_vel_gain");
```
Comment (satoshi-ota) [inline vehicle/external_cmd_converter/src/node.cpp:45]:
```suggestion
  ref_vel_gain_ = declare_parameter<double>("ref_vel_gain");
```

File: vehicle/external_cmd_converter/src/node.cpp [hunk 3]
```diff
@@ -42,13 +42,13 @@ ExternalCmdConverterNode::ExternalCmdConverterNode(const rclcpp::NodeOptions & n
...
 
   // Parameter for Hz check
-  const double timer_rate = declare_parameter("timer_rate", 10.0);
-  wait_for_first_topic_ = declare_parameter("wait_for_first_topic", true);
-  control_command_timeout_ = declare_parameter("control_command_timeout", 1.0);
-  emergency_stop_timeout_ = declare_parameter("emergency_stop_timeout", 3.0);
+  const double timer_rate = declare_parameter<number>("timer_rate");
+  wait_for_first_topic_ = declare_parameter<boolean>("wait_for_first_topic");
```
Comment (satoshi-ota) [inline vehicle/external_cmd_converter/src/node.cpp:49]:
```suggestion
  wait_for_first_topic_ = declare_parameter<bool>("wait_for_first_topic");
```

File: vehicle/external_cmd_converter/src/node.cpp [hunk 4]
```diff
@@ -42,13 +42,13 @@ ExternalCmdConverterNode::ExternalCmdConverterNode(const rclcpp::NodeOptions & n
...
+  control_command_timeout_ = declare_parameter<number>("control_command_timeout");
```
Comment (satoshi-ota) [inline vehicle/external_cmd_converter/src/node.cpp:50]:
```suggestion
  control_command_timeout_ = declare_parameter<double>("control_command_timeout");
```

File: vehicle/external_cmd_converter/src/node.cpp [hunk 5]
```diff
@@ -42,13 +42,13 @@ ExternalCmdConverterNode::ExternalCmdConverterNode(const rclcpp::NodeOptions & n
...
+  emergency_stop_timeout_ = declare_parameter<number>("emergency_stop_timeout");
```
Comment (satoshi-ota) [inline vehicle/external_cmd_converter/src/node.cpp:51]:
```suggestion
  emergency_stop_timeout_ = declare_parameter<double>("emergency_stop_timeout");
```

**Comment by satoshi-ota** [comment]
@karishma1911 Sorry. Could you fix build errors?

**Comment by karishma1911** [comment]
> @karishma1911 Sorry. Could you fix build errors?

yes, i fixed above errors

**Pull request merged**